src/main/java/com/google/firebase/remoteconfig/internal/TemplateResponse.java

```diff
@@ -0,0 +1,123 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.remoteconfig.internal;
+
+import com.google.api.client.util.Key;
+import com.google.firebase.remoteconfig.RemoteConfigParameter;
+import com.google.firebase.remoteconfig.RemoteConfigParameterValue;
+import com.google.firebase.remoteconfig.RemoteConfigTemplate;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * The DTO for parsing Remote Config template responses from the Remote Config service.
+ **/
+public final class TemplateResponse {
+
+  @Key("parameters")
+  private Map<String, ParameterResponse> parameters;
+
+  public TemplateResponse() {
+    parameters = Collections.emptyMap();
+  }
+
+  public TemplateResponse(Map<String, ParameterResponse> parameters) {
+    this.parameters = parameters;
+  }
+
+  public RemoteConfigTemplate toRemoteConfigTemplate() {
+    Map<String, RemoteConfigParameter> parameterPublicTypes = new HashMap<>();
+    for (Map.Entry<String, ParameterResponse> entry : parameters.entrySet()) {
+      parameterPublicTypes.put(entry.getKey(), entry.getValue().toRemoteConfigParameter());
+    }
+    return new RemoteConfigTemplate().setParameters(parameterPublicTypes);
+  }
+
+  /**
+   * The DTO for parsing Remote Config parameter responses from the Remote Config service.
+   **/
+  public static final class ParameterResponse {
+
+    @Key("defaultValue")
+    private ParameterValueResponse defaultValue;
+
+    @Key("description")
+    private String description;
+
+    @Key("conditionalValues")
+    private Map<String, ParameterValueResponse> conditionalValues;
+
+    public ParameterResponse() {
+      conditionalValues = Collections.emptyMap();
+    }
+
+    public ParameterResponse(ParameterValueResponse defaultValue, String description,
+                             Map<String, ParameterValueResponse> conditionalValues) {
+      this.defaultValue = defaultValue;
+      this.description = description;
+      this.conditionalValues = conditionalValues;
+    }
+
+    public RemoteConfigParameter toRemoteConfigParameter() {
+      Map<String, RemoteConfigParameterValue> conditionalPublicValues = new HashMap<>();
+      for (Map.Entry<String, ParameterValueResponse> entry : conditionalValues.entrySet()) {
+        conditionalPublicValues
+                .put(entry.getKey(), entry.getValue().toRemoteConfigParameterValue());
+      }
+      return new RemoteConfigParameter()
+              .setDefaultValue(defaultValue.toRemoteConfigParameterValue())
+              .setDescription(description).setConditionalValues(conditionalPublicValues);
+    }
+  }
+
+  /**
+   * The DTO for parsing Remote Config parameter value responses from the Remote Config service.
```

Comment:
I wonder if we should spell out "Data Transfer Object" -- that's what this is, right?  :)

I searched and couldn't immediately find a single other instance of "DTO" in docs/reference/

Reply:
I think the classes in `internal` packages do not show up in the docs. I agree though `Data Transfer Object` makes more sense. I will update the comment here. Thanks!!